### ISSUE TITLE
create an outline / plan for a gcdt getting-started guide

### ISSUE THREAD

**Issue by markfink** [glomex/gcdt#312]
**what would be the most useful gcdt getting-started guide**

write the outline from the perspective of a DevOps developer new to gcdt... what would be the most helpful.

probably a good idea to do a little research on google to see if some useful "getting started guides" come up on the topic of "AWS deployment" ... I do not think we have to reinvent the wheel if a guide with a perfect structure already exists - I believe the clever move would be to reuse a good and proven structure.

Please let us review the outline before you proceed to writing the actual getting-started guide. Please attach new revisions of the outline / plan to this issue if possible or submit as PR.

@jbrandstetter  on the topic of getting-started / supercars docu:

"the overall topic would be that after you understand all the different gcdt tools and their inner workings you should also be able to contribute code if the need comes up in your squad. also we’d like you to then start working on a detailed user’s guide from a beginner’s view. (quick start, tutorial, etc) because most of us already have some cognitive bias ;)"

**intended audience**

* developer
* knows AWS
* knows the service he / she is working on
* understands deployments in general

**requirements**

* use supercars as a sample (https://github.com/glomex/gcdt/issues/311)
* implement the `gcdt getting-started guide` as part of gcdt docs (http://gcdt.readthedocs.io/en/latest/)
* use the gcdt issues and Project kanban board to track progress (https://github.com/glomex/gcdt/projects/1)
* one of the most important things for the getting-started guide IMHO is the concept of infrastructure-as-code. Consequently it must put emphasis on this concept and revisit it frequently i.e. what it means in the context of kumo, tenkai, ramuda and yugen. I believe infrastructure-as-code is kind of a red-thread leading through the getting-started guide which shows how gcdt explicit infrastructure configuration makes it possible to apply the same tools and concepts we are  already familiar with and use to manage software and now apply them to working with infrastructure.
* please, in case you are missing some gcdt docu or the existing docu is incomplete or difficult to comprehend please drop a line with your suggestion into https://github.com/glomex/gcdt/issues/221
* on 30.06 we agreed to call this `Getting starated guide` and the file `03_getting_started.md`

**Comment by markfink** [comment]
proposal 2017-06-27 by max:
```
1) gcdt installation. Write about how to instal using pip and requirements files.
2) Small overview of all gcdt services: kumo, tenkai, ramuda, yugen.
3) Provide overview of kumo service. Give an examples of simple stack using python and
troposphere.
  - write cloudformation.py to deploy some simple stack(EC2, ASG, Security group, ELB) to
  deploy for example LAMP stack
4) Provide overview of tenkai. Give an examples. Explain how to use appspec.yml and how to
verify service installation.
  - Write simple code deploy script to start LAMP stack and verify that it work
5) Provide overview of yugen. Give an examples.
```
@markfink recommended two follow up actions:
1) please put your plan into a gcdt issue (https://github.com/glomex/gcdt/issues/312) 
2) I think this is a valid outline but not anywhere detailed enough. so please provide more details (especially on the structure)

**Comment by jbrandstetter** [comment]
@markfink re on the topic of quickstart / supercars docu:
this "the overall topic would be that after you understand all the different gcdt tools and their inner workings you should also be able to contribute code if the need comes up in your squad. " was the intended goal for maksym (or any new SRE) after leaving our squad to join their destination squad. So every SRE should be able to contribute to gcdt. For the documentation this means that we maybe at some point need some sort of "contribution guide" but not now. Let's focus on documentation (examples, guides, etc) for beginners.

**Comment by Dgadavin** [comment]
Here is the basic [structure](https://github.com/glomex/gcdt/blob/beginners-guide-doc/docs/gcdt_beginners_guide/02_structure.md). If it's Ok, I will continue filling it with info

**Comment by markfink** [comment]
@Dgadavin I think this structure is pretty straight forward [structure](https://github.com/glomex/gcdt/blob/beginners-guide-doc/docs/gcdt_beginners_guide/02_structure.md).
The only thing I would add in the beginning is some motivation on why we automate and why automation is important. I believe an introduction to IoC is essential. We could add this later but I think it will come out better if you use this as a building block. @jbrandstetter what do you think about the outline?
Also you need to explain some plugin stuff... probably part of the installation but you should briefly cover the motivation for the plugin like customisation for your project and your tools (slack, datadog, git, config formats) . hope that makes sense

**Comment by jbrandstetter** [comment]
So far so good. I agree with Mark that we should explain the motivation for the plugin mechanism, explain the installation in detail and also very important have a best practices section covering Python on Mac, virtualenv, pip - basically the top 10 problems people always have. I'd also add something for the impatient, like "gcdt in 5 minutes"

**Comment by Dgadavin** [comment]
I've opened #315 PR. We can start reviewing this userguide partially. Installation section complete from my perspective. 
Would appreciate any feedbacks. @markfink @jbrandstetter

**Comment by Dgadavin** [comment]
Closed by #315
